packages/fast-usdc/src/utils/fees.js

```diff
@@ -0,0 +1,57 @@
+import { AmountMath } from '@agoric/ertp';
+import { multiplyBy } from '@agoric/zoe/src/contractSupport/ratio.js';
+import { Fail } from '@endo/errors';
+import { mustMatch } from '@endo/patterns';
+import { FeeConfigShape } from '../type-guards.js';
+
+const { add, isGTE, min, subtract } = AmountMath;
+
+/**
+ * @import {Amount} from '@agoric/ertp';
+ * @import {FeeConfig} from '../types.js';
+ */
+
+/** @param {FeeConfig} feeConfig */
+export const makeFeeTools = feeConfig => {
+  mustMatch(feeConfig, FeeConfigShape, 'Must provide feeConfig');
+  const { flat, variableRate, maxVariable } = feeConfig;
+  const feeTools = harden({
+    /**
+     * Calculate the net amount to advance after withholding fees.
+     *
+     * @param {Amount<'nat'>} requested
+     * @throws {Error} if fee exceeds requested
+     */
+    calculateAdvance(requested) {
+      const fee = feeTools.calculateAdvanceFee(requested);
+      return subtract(requested, fee);
+    },
+    /**
+     * Calculate the total fee to charge for the advance.
+     *
+     * @param {Amount<'nat'>} requested
+     * @throws {Error} if fee exceeds requested
+     */
+    calculateAdvanceFee(requested) {
+      const variable = min(multiplyBy(requested, variableRate), maxVariable);
+      const fee = add(variable, flat);
+      !isGTE(fee, requested) || Fail`Fee exceeds requested.`;
```

Comment:
If `fee == requested` then isGTE will be true, so !isGTE will be false and the Fail will throw.

```suggestion
      isGTE(requested, fee) || Fail`Fee exceeds requested.`;
```

Or,

```suggestion
      !isGTE(fee, requested) || Fail`Request must exceed fees.`;
```

If the latter, updates the `@throws` too.

Reply:
Updated the error message and `@throws`